### ISSUE TITLE
Help, what could be the possible reasons for image distortion?

### ISSUE THREAD

**Issue by Asy-n** [mpetroff/pannellum#1247]
First of all, thank you very much to the author for the open source project! I am facing an issue right now.

I can access the image normally when using front-end resources, but after uploading it to the backend server, when I try to move the view a second time, the image distorts into concentric circles.

Does the image resource need to be returned by a specific service? The backend stores the images in a static resource folder, and I am accessing the folder directly via the backend domain.

normal
![image](https://github.com/user-attachments/assets/86173515-1ef2-4168-aa3d-f487bdb049b5)

error
![image](https://github.com/user-attachments/assets/e4cb921b-87fc-482a-ad7b-f2448160343b)

**Comment by mpetroff** [comment]
I've never seen anything that's looked quite like that before. Are there any errors printed to the browser console? Have you tried loading it on different computers or different browsers?

> Does the image resource need to be returned by a specific service?

As long as the image can be loaded with JavaScript (e.g., you don't have CORS issues), there's nothing specific.

**Comment by Asy-n** [comment]
Apologies for the late response. 
The console does not report any errors when the image distorts into a concentric circle. I have tried different browsers and computers, and the issue remains the same. 
During debugging, I placed the image in two different locations, both under the ng domain pointing to folders. 
One folder is the Vue project where I placed the image in the public directory, and it rotates normally when placed here. The other is a regular folder with no service, just a static folder. When requesting the image resource from this location, the image becomes distorted. 
I suspected that it might be due to ng configuring gzip, but the header information and returned image sizes of requests from both paths are consistent.
![image](https://github.com/user-attachments/assets/f28d23d4-ed69-4242-a2e6-cd4bdbbc79d4)
![image](https://github.com/user-attachments/assets/1a85b6f4-34f3-4b14-9958-8430b9627776)
The browser shows these warnings, but they also appear when the image rotates normally.

**Comment by mpetroff** [comment]
While I've never seen perfect concentric circles before, I have seen stripes before. The issue this reminded me of was #323, although it's unrelated, as it was caused by some sort of GPU driver issue, which clearly isn't the case here, since it works for you in some cases and is persistent across devices. In searching for that issue, I did find #852, which shows partial circles. Thinking about it more, I was able to reproduce your circles, by setting `yaw` to `NaN` and pitch to `-90` (the `-90` pitch is what turns the stripes into circles). I also found #757, which while I didn't identify the issue at the time, had similar artifacts caused by setting `yaw` to a `String` (since that's what a URL parameter is) instead of a `Number`, as required.

Based on this and the fact that you reported that the issue only arises when you try to move the view, I don't think it has anything to do with where the image is hosted. I suspect something else is changing in your web app, although somehow triggered by the changing the image hosting location.

What methods have you tried to use to rotate the image? Your screenshots appear to be from a phone, so I assume you've used touch input, but have you also tried to use mouse input or keyboard input?

Do the `getYaw()` and `getPitch()` methods return `NaN` and `-90`, respectively, as I'm hypothesizing is the case?

**Comment by Asy-n** [comment]
> 虽然我之前从未见过完美的同心圆，但我之前见过条纹。这让我想起了#323这个问题，尽管它与此无关，因为它是由某种GPU驱动程序问题引起的，但是显然不是这里的情况，因为它在某些情况下对您有用并且在不同的设备之间是持久的。在搜索该问题时，我确实找到了#852，它显示了部分订单。之后进一步考虑，我可以通过将和距离设置为重新来您的发票`yaw`（退货变成了发票的原因）。我还发现了#757，虽然我当时没有发现问题，但它有类似的瑕疵，这是由于按要求将其设置为`NaN`一个（因为这就是 URL 参数）而不是一个相反导致的。`-90``-90`[](https://github.com/mpetroff/pannellum/issues/757)`yaw``String``Number`
> 
> 基于这一点以及您报告的问题仅在您尝试移动视图时出现的事实，我认为这与图片托管位置相关。我怀疑您的 Web 应用程序中还有其他东西正在发生变化，尽管这是由更改的图片引起的托管位置以某种方式触发的。
> 
> 您尝试使用什么方法来旋转图像？您的屏幕截图似乎来自手机，所以我假设您使用了触摸输入，但您是否也尝试使用鼠标输入或键盘输入？
> 
> 正如我想象的那样，`getYaw()`和方法是否分别`getPitch()`返回`NaN`和？`-90`
Indeed, the error was caused by the String type. The default view of my local images was set to the Number type 0/0 by myself. After uploading the materials to the backend, the backend stored Yaw and Pitch as String, which led to this distortion error. I really appreciate your help in answering my questions. Your tool has been very helpful to me. Wishing you all the best in the new year!
